cfgov/v1/util/queries.py

```diff
@@ -0,0 +1,52 @@
+import logging
+import ref
+
+from ..models.learn_page import AbstractFilterPage
+
+logger = logging.getLogger(__name__)
+
+
+class FilterableQuery(object):
+
+    def __init__(self, form, hostname, page=None):
+        self.form = form
+        self.hostname = hostname
+        self.page = page
+
+    def update_categories(self):
+        """ This is a (hopefully) temporary solution for dealing w/ the fact
+        that we show Blog and Reports as options for filtering, but
+        aren't categories themselves. Rather, they consist of sub-categories,
+        but since we aren't showing those sub-categories to the end user,
+        selecting the parent category is equivalent to all of them
+        getting checked. The mapping of Blog and Reports to their
+        respective categories exists in cfgov/v1/util/ref.py
+        """
+        categories = self.form.cleaned_data.get('categories', [])
+        if 'blog' in categories:
+            self.form.cleaned_data['categories'].remove('blog')
+            for x in ref.categories[1][1]:
+                self.form.cleaned_data['categories'].append(x[0].lower())
+        if 'research-reports' in categories:
+            self.form.cleaned_data['categories'].remove('research-reports')
+            for x in ref.categories[8][1]:
+                self.form.cleaned_data['categories'].append(x[0].lower())
+
+    def get_page_set(self):
+        if self.form.__class__.__name__ == 'FilterableListForm':
```

Comment:
I don't quite understand why this is the relevant check for whether to include the parent page in the filter or not. Is that just an unfortunate side effect of the design that we only use `FilterableListForm` in that case, but children of `FilterableListForm` otherwise?

Could you instead check whether `page == None`, and only pass it into this class if you want to filter by the parent page? Am I interpreting it right that `self.page` here means "parent page, only return pages that are children of this one" -- if so, would it make sense to name that as `self.parent_page`?